telink: tlx: ieee802154:  power consumption optimization

liang4804 opened the pull request:
- add wfi to optimize the power consumption for tlx platform.
- move the rf buffer from IRAM to DRAM for tl321x.

liang4804 committed:
zephyr: for CI check

move the net buffer from iram to dram

Signed-off-by: Liang Zhong <liang.zhong@telink-semi.com>

liang4804 committed:
tl321x_retention: add some config for tl321x_retention

add one more power mode and set the clock 96M for tl321x_retention

Signed-off-by: Liang Zhong <liang.zhong@telink-semi.com>

liang4804 committed:
tlx321x_retention: 802154 optimization

add wfi to optimize the current of tlx.

move the rf buffer from iram to dram.

Signed-off-by: Liang Zhong <liang.zhong@telink-semi.com>

damien0x0023 commented:
@liang4804
PRs #419 and #420 seem to be duplicates of this PR.
@wuhaoTelink 
We may close other PRs but work on this one using the _develop_tlx802154_optimization_ branch for verification and refinement.

damien0x0023 committed:
telink: tlx: ieee802154: add Thread state flag

- introduce new bool isThreadCommissioned
to optimize power consumption during ieee802154.
- remove unnecessary B9X condition under tlx files.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>
